File: Assigment 3/C/02. store (1 to 9) in a matrix in spiral manner .c
// 1 2 3
// 8 9 4
// 7 6 5

#include <stdio.h>

int main(){
    int m,n;
    printf("enter row: ");
    scanf("%d",&m);
    printf("enter column: ");
    scanf("%d",&n);

    int arr[m][n];

    printf("array = \n");

    int top = 0;
    int bottom = m-1;
    int left = 0;
    int right = n-1;

    int a = 1;

    while(top<=bottom && left<=right){
        for(int i=left; i<=right; i++){
            arr[top][i] = a++;
        }
        top++;

        for(int i=top; i<=bottom; i++){
            arr[i][right] = a++;
        }
        right--;

        if(top<=bottom){
            for(int i=right; i>=left; i--){
                arr[bottom][i] = a++;
            }
            bottom--;
        }

        if(left<=right){
            for(int i=bottom; i>=top; i--){
                arr[i][left] = a++;
            }
            left++;
        }
    }

    for(int i=0; i<m; i++){
        for(int j=0; j<n; j++){
            printf("%d ",arr[i][j]);
        }
        printf("\n");
    }

    return 0;
}

// steps :

// t0 r
// l0 b

// l --> r
// t --> b 

// r --> l 
// b --> t 

File: Assigment 3/Python/02. store (1 to 9) in a matrix in spiral manner .py
# 1 2 3
# 8 9 4
# 7 6 5

m = int(input("enter row: "))
n = int(input("enter column: "))

arr = [[0]*n for _ in range(m)]

top = 0
left = 0
bottum = m-1
right = n-1

a=1;
while top<=bottum and left<=right:
    for i in range(left,right+1):
        arr[top][i] = a
        a+=1
    top+=1

    for i in range(top,bottum+1):
        arr[i][right] = a
        a+=1
    right-=1

    if(top<=bottum):
        for i in range(right,left-1,-1):
            arr[bottum][i] = a
            a+=1
        bottum-=1   
    if(left<=right):
        for i in range(bottum,top-1,-1):
            arr[i][left] = a
            a+=1
        left+=1

for i in range(m):
    for j in range(n):
        print(arr[i][j],end=" ")
    print("\n")

